services/navigation/navigation.go

```diff
@@ -74,28 +74,92 @@ func StringToMapType(mapTypeName string) (MapType, error) {
 	return 0, errors.Errorf("invalid map_type '%v' given", mapTypeName)
 }
 
-// Properties returns information regarding the current navigation service. This includes the map type
+// Properties returns information about the MapType that the configured navigation service is using.
 // being ingested and used by the navigation service.
 type Properties struct {
 	MapType MapType
 }
 
 // A Service controls the navigation for a robot.
+//
+// Mode example:
+//	// Get the Mode the service is operating in
+//	mode, err := myNav.Mode(context.Background(), nil)
+//
+// SetMode example:
+//	// Set the Mode the service is operating in to ModeWaypoint and begin navigation
+//	err := myNav.SetMode(context.Background(), navigation.ModeWaypoint, nil)
+//
+// Location example:
+//	// Get the current location of the robot in the navigation service
+//	location, err := myNav.Location(context.Background(), nil)
+//
+// Waypoints example:
+//	waypoints, err := myNav.Waypoints(context.Background(), nil)
+//
+// AddWaypoint example:
+//	// Create a new waypoint with latitude and longitude values of 0 degrees
+//	// Assumes you have imported "github.com/kellydunn/golang-geo" as `geo`
+//	location := geo.NewPoint(0, 0)
+//
+//	// Add your waypoint to the service's data storage
+//	err := myNav.AddWaypoint(context.Background(), location, nil)
+//
+// RemoveWaypoint example:
+//	// Assuming you have already called AddWaypoint once and the waypoint has not yet been reached
+//	waypoints, err := myNav.Waypoints(context.Background(), nil)
+//	if (err != nil || len(waypoints) == 0) {
+//	    return
+//	}
+//
+//	// Remove the first waypoint from the service's data storage
+//	err := myNav.RemoveWaypoint(context.Background(), waypoints[0].ID, nil)
+//	err := myNav.RemoveWaypoint(context.Background(), waypoint_id, nil)
+//
+// Obstacles example:
+//	// Get an array containing each obstacle stored by the navigation service
+//	obstacles, err := myNav.Obstacles(context.Background(), nil)
+//
+// Paths example:
+//	// Get an array containing each path stored by the navigation service
+//paths, err := myNav.Paths(context.Background(), nil)
+//
+// Properties example:
+//
+//	// Get the properties of the current navigation service
+//	navProperties, err := myNav.Properties(context.Background())
```

Comment:
[nit]: please add a space below the examples and above `type Service interface {` just for formatting

Reply:
please add a space below here so there is a slight separation between our examples and `type Service interface {`

Note: in case this is not possible because of how our linter works that is ok :)

Reply:
> if you know a workaround or alternative! Until then I will revert it.

if its failing lint it ok to remove imo, no biggie.
We can also run `make lint-go build-go` from `rdk`. This will make changes to your files which you can then commit so that there is no mismatch between the CI and what you have locally.